app/code/Magento/Inventory/Ui/Component/Control/Stock/GenericButton.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Ui\Component\Control\Stock;
+
+use Magento\Backend\Block\Widget\Context;
+use Magento\InventoryApi\Api\Data\StockInterface;
+
+/**
+ * Class GenericButton
+ */
+class GenericButton
+{
+    /**
+     * @var \Magento\Framework\UrlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param Context $context
+     */
+    public function __construct(Context $context)
+    {
+        $this->urlBuilder = $context->getUrlBuilder();
+        $this->context = $context;
+    }
+
+    /**
+     * Get stock id
+     *
+     * @return int|null
+     */
+    public function getId()
+    {
+        return $this->context->getRequest()->getParam(StockInterface::STOCK_ID) ?? null;
```

Comment:
This code 
`$this->context->getRequest()->getParam(StockInterface::STOCK_ID) `
violates the Law of Demeter https://en.wikipedia.org/wiki/Law_of_Demeter
which is one of the good design guidelince for developing Object Oriented software